src/linker/implementation.py

```diff
@@ -0,0 +1,60 @@
+import os
+from pathlib import Path
+from typing import List
+
+from linker.configuration import Config
+from linker.utilities.general_utils import load_yaml
+
+
+class Implementation:
+    def __init__(self, step_name: str, config: Config, allowable_implementations: List[str]):
+        self.step_name = step_name
+        self.config = config
+        self.allowable_implentations = allowable_implementations
+
+    def __repr__(self):
+        return f"{self.step_name}_implementation"
+
+    @property
+    def implementation_dir(self) -> Path:
+        # TODO: move this into proper config validator
+        implementation = self.config.pipeline["steps"][self.step_name]["implementation"]
```

Comment:
I think an implementation should know its implementation_dir on initialization, and not be referencing these other objects in order to get it.

Reply:
Separate PR